chainer/functions/evaluation/classification_summary.py

```diff
@@ -90,6 +90,9 @@ def classification_summary(y, t, label_num=None, beta=1.0, ignore_label=-1):
     into account for calculating the above quantities.
     By default, it is set to -1 so that all instances are taken
     into consideration, as labels are supposed to be non-negative integers.
+    If ``ignore_label`` is "valid", that is, it is a non-negative
+    integer less than ``label_num``, then the ``ignore_label``-th
+    entries in output arrays do not contain correct quantities.
```

Comment:
I'm not sure if my understanding w.r.t. your comment is correct, but it seems that you have considered that setting ``ignore_label`` to a value less than ``n_classes`` is **illegal** in this function.

My opinion is the following; it is the best for us to completely support ``ignore_label`` within {0, ..., n_label}, but it requires to deal with the problem that setting a small non-negative value raises ``RuntimeWarning``. Therefore, currently we define that setting a non-negative integer less than ``n_label`` to``ignore_label`` is "incorrect", and yields "undefined behavior" (even in this setting this function works well to a certain degree, though).
In this problem we should solve it by some PRs, but it is a future-work.